### PR TITLE
Added my workshop on pure JavaScript development for API data 

### PR DESCRIPTION
**Pull request by Taiwrash**
This is the link to recording of the workshop on slack: https://hackclub.slack.com/archives/C01A9V5LPD0/p1603065621012500

I am Mudasiru Rasheed and I am Rasheed on the slack channel also leading hackclub-futminna.

### CODE REVIEW

File: workshops/country-list-app/README.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
+---
+name: "World Countries and their Flags with Javascript"
+description: "This is an app that contain the list of all the countries in the world with teir names and flags"
```
Comment (bezlin6mechminerz) [inline workshops/country-list-app/README.md:3]:
```suggestion
description: "This is an app that contains the list of all the countries in the world with their names and flags"
```

File: workshops/country-list-app/README.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+author: "@Taiwrash"
+---
+
+## Country List
+
+![Sample Page](https://cloud-g6yi6cyg3.vercel.app/0untitled.png)
+
+This is the [Live Demo](https://country-list.taiwrash.repl.co) of the app and [Full Code](https://repl.it/@Taiwrash/country-list).
+
+_We recomend any browser and good network andor internet access._
```
Comment (bezlin6mechminerz) [inline workshops/country-list-app/README.md:13]:
```suggestion
_We recommend any browser and good network access._
```

File: workshops/country-list-app/README.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    <body>
+      <main id="main">
+        <div class="card"></div>
+      </main>
+      <script src="script.js"></script>
+    </body>
+  </html>
+```
+
+**NOTE:** `You can copy paste the code above` But, never copy paste a non-clear code or the code you did not understand. A brief explanation of the html boilerplate at the top is as simple as having `css` file linked at the `head` section of the html along with the title element to show web title and the main element for the purpose of semantic and web page accessibilties. Followed by a div with a class attribute of card which shall hold every country flags and names and lastly linked with a javascript `script` file that will do all the magic, just before the body closing tag.
```
Comment (bezlin6mechminerz) [inline workshops/country-list-app/README.md:48]:
```suggestion
**NOTE:** `You can copy-paste the code above` But, never copy paste a non-clear code or the code you did not understand. A brief explanation of the html boilerplate at the top is as simple as having `css` file linked at the `head` section of the html along with the title element to show web title and the main element for the purpose of semantic and web page accessibilities. Followed by a div with a class attribute of card which shall hold every country flags and names and lastly linked with a javascript `script` file that will do all the magic, just before the body closing tag.
```

File: workshops/country-list-app/README.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+```
+  * {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+  }
+```
+
+This is purposely to remove the default margin and padding of every element and to set the box-sizing to border-box which controls behaviour activities of every elements on the page. This prevent default addition of margin and padding to the width and height.
```
Comment (bezlin6mechminerz) [inline workshops/country-list-app/README.md:62]:
```suggestion
This is purposely to remove the default margin and padding of every element and to set the box-sizing to border-box which controls behavior activities of every element on the page. This prevents the default addition of margin and padding to the width and height.
```

File: workshops/country-list-app/README.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+  width: 100vw;
+  background-color: gainsboro;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  padding-top: 50px;
+  }
+```
+
+Both the width and height are set to the view width and view height of the viewing device. It is display flex (Flex is a new feature in css that encourage and improve responsiveness of every web pages) you can read more at [Read more Here](https://www.google.com/url?sa=t&source=web&rct=j&url=https://www.freecodecamp.org/news/flexbox-the-ultimate-css-flex-cheatsheet/amp/&ved=2ahUKEwiX0LzIrL_sAhWsyIUKHRveA2EQFjAkegQIGhAB&usg=AOvVaw1L3IU0DUtdahRsgB-n5vBS&ampcf=1)
```
Comment (bezlin6mechminerz) [inline workshops/country-list-app/README.md:77]:
```suggestion
Both the width and height are set to the view width and view height of the viewing device. It is display flex (Flex is a new feature in css that encourage and improve the responsiveness of every web pages) you can read more at [Read More Here](https://www.google.com/url?sa=t&source=web&rct=j&url=https://www.freecodecamp.org/news/flexbox-the-ultimate-css-flex-cheatsheet/amp/&ved=2ahUKEwiX0LzIrL_sAhWsyIUKHRveA2EQFjAkegQIGhAB&usg=AOvVaw1L3IU0DUtdahRsgB-n5vBS&ampcf=1)
```

File: workshops/country-list-app/README.md [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+        countryName.textContent = item.name
+
+        cardParent.appendChild(flagImage);
+        cardParent.appendChild(countryName);
+      }
+     });
+    });
+```
+
+**_ This is the code that did all the magic and I will be expalining what each section is doing one after the other _**
```
Comment (bezlin6mechminerz) [inline workshops/country-list-app/README.md:153]:
```suggestion
**_ This is the code that did all the magic and I will be explaining what each section is doing one after the other _**
```

File: workshops/country-list-app/README.md [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+
+Then javascript fetch method was invoked with the argument/parameters of base url.
+
+### What is Fetch?
+
+fetch is one of the way of making `HTTP` calls. http calls are numerous ranging from `GET`, `POST`, `DELETE` among nine (9) others which are not the scope of this `WORKSHOP`. Fetch method return a promise.
+
+### What is a Promise?
+
+Promise is that object that may produce single value in the future. Response which denote fulfilment or rejection that is success or failue which is handle in javascript with `.then() method`.
```
Comment (bezlin6mechminerz) [inline workshops/country-list-app/README.md:192]:
```suggestion
The promise is that object that may produce a single value in the future. A response that denotes fulfilment or rejection that is success or failure which is handle in javascript with `.then() method`.
```

**Comment by bezlin6mechminerz** [comment]
Hey! There is spelling and grammer mistakes in this workshop i just pointed out some. Just figure it out.

**Comment by faisalsayed10** [comment]
@bezlin6mechminerz I've told you once, please follow the rules.

**Review by Taiwrash — COMMENTED** [review]
Thanks so much for the quick response and review. You mentioned me following rules, I actually do my possible best to adhere to workshop guide as laid down by hackclub. I will very happy if you can suggest certain rules I violated to make corrections. Thanks so I really appreciate @bezlin6mechminerz

**Comment by MatthewStanciu** [comment]
Thanks for submitting this workshop! I'll review it soon.

**Comment by MatthewStanciu** [comment]
Hey, thanks for submitting this! Unfortunately, we can't accept this workshop because it doesn't follow several of the rules laid out in the [submission requirements](https://hackclub.com/workshop-bounty):

1. The workshop doesn't appear to be finished—it seems like it ends abruptly and doesn't leave the user feeling like they completed a project
2. The project isn't easily hackable, and doesn't list 3 ways to take it further at the end
3. Some of the grammar and markdown formatting is a little off

I'm going to close this for now, but if you want to resubmit this workshop following all of the submission guidelines, feel free to do so and I'll re-review.

**Pull request closed without merging**